tests/test_sharing_client.py

```diff
@@ -0,0 +1,229 @@
+                mock_refresh_sharing_keys.return_value = EncryptionKeysCollection([master_key, site_key],
+                                                                                  expected_scope, site_id, 1,
+                                                99999, 86400, max_sharing_lifetime)
+                self._client.refresh_keys()
+                with self.assertRaises(EncryptionError) as context:
+                    self._client.decrypt_token_into_raw_uid(token)
+                self.assertEqual('invalid token lifetime', str(context.exception))
+
+    def test_token_generated_in_the_future_to_simulate_clock_skew(self, mock_refresh_sharing_keys):  # TokenGeneratedInTheFutureToSimulateClockSkew
+        created_at_future = dt.datetime.now(tz=timezone.utc) + dt.timedelta(minutes=31)  #max allowed clock skew is 30m
+        for expected_scope, expected_version in test_cases_all_scopes_all_versions:
+            with self.subTest(expected_scope=expected_scope, expected_version=expected_version):
+                token = generate_uid_token(expected_scope, expected_version, created_at=created_at_future)
+                mock_refresh_sharing_keys.return_value = EncryptionKeysCollection([master_key, site_key],
+                                                                                  expected_scope, site_id, 1,
+                                                99999, 86400)
+                self._client.refresh_keys()
+                with self.assertRaises(EncryptionError) as context:
+                    self._client.decrypt_token_into_raw_uid(token)
+                self.assertEqual('invalid token lifetime', str(context.exception))
+
+    def test_token_generated_in_the_future_within_allowed_clock_skew(self, mock_refresh_sharing_keys):  # TokenGeneratedInTheFutureWithinAllowedClockSkew
+        created_at_future = dt.datetime.now(tz=timezone.utc) + dt.timedelta(minutes=29)  #max allowed clock skew is 30m
+        for expected_scope, expected_version in test_cases_all_scopes_all_versions:
+            with self.subTest(expected_scope=expected_scope, expected_version=expected_version):
+                token = generate_uid_token(expected_scope, expected_version, expires_at=created_at_future)
+                mock_refresh_sharing_keys.return_value = EncryptionKeysCollection([master_key, site_key],
+                                                                                  expected_scope, site_id, 1,
+                                                99999, 86400)
+                self._client.refresh_keys()
+                result = self._client.decrypt_token_into_raw_uid(token)
+                self.assertIsNotNone(result)
+                self.assertEqual(result.identity_scope, expected_scope)
+                self.assertEqual(result.advertising_token_version, expected_version)
+
+    def test_phone_uids(self, mock_refresh_sharing_keys):  # PhoneTest
```

Comment:
Is this the PhoneTest equivalent? Let's use the same order between SDKs to make it easier

Reply:
I actually did reorganize the test cases between the SDKs :)

Reply:
Could you elaborate? We need to keep them in the same order to make it easier for future maintainers. If you want to propose a better order, let's discuss but you'll need to change them in all SDKs.

Reply:
I just meant that I sync'd the tests in python SDK with the C# one. They were not in the same order in Python SDK.